### PR TITLE
Upgrade react-native-push-notification

### PR DESCRIPTION
**Pull request by OzzieOrca**
- Remove unneeded GCM_SENDER_ID which should be pulled from native firebase config
- Fixes https://jira.cru.org/browse/MHP-3365 by setting `requestPermissions` to `true` on Android

~I was struggling getting any push notifications a day or two ago and did this. Didn't really solve anything but might be good to still merge?~

### CODE REVIEW

**Review by TheNoodleMoose — DISMISSED** [review]
So you're still unable to get push notifications?

File: src/actions/notifications.ts [hunk 1]
```diff
@@ -256,11 +255,8 @@ export function configureNotificationHandler() {
         !__DEV__ && rollbar.error(error);
       },
 
-      // ANDROID ONLY: GCM Sender ID
-      senderID: GCM_SENDER_ID,
-
-      // we manually call this after to have access to a promise for the iOS prompt
-      requestPermissions: false,
+      // we manually call this after on iOS to have access to a promise for the iOS prompt
+      requestPermissions: isAndroid,
```
Comment (OzzieOrca) [inline src/actions/notifications.ts:259]: I think this is what fixed things. Thanks for figuring that out!

Comment (TheNoodleMoose) [reply src/actions/notifications.ts:259]: Yay, glad this worked 🙂

File: .env.production [hunk 2]
```diff
@@ -1,5 +1,4 @@
 API_BASE_URL=https://api.missionhub.com
-GCM_SENDER_ID=208966923006
```
Comment (TheNoodleMoose) [inline .env.production:2]: What was this used for exactly?

**Review by TheNoodleMoose — APPROVED** [review]

File: .env.production (hunk 2 above)
Comment (OzzieOrca) [reply .env.production:2]: I think an old version needed it. Maybe GCM before it migrated to FCM. My comment above says it's "pulled from native firebase config" now so my old self must have read something about that... It's missing from the new typescript types and the docs now.

File: src/actions/notifications.ts (hunk 1 above)
Comment (OzzieOrca) [reply src/actions/notifications.ts:259]: Just wanted to document things. This fix was originally discussed in https://github.com/CruGlobal/missionhub-react-native/pull/2019.

**Pull request merged**